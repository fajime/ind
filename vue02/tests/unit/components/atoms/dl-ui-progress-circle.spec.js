import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import DlUiProgressCircle from '../../../../src/components/atoms/dl-ui-progress-circle/index.vue';

describe('dl-ui-progress-circle', () => {

  const wrapper = mount(DlUiProgressCircle);

  it('Component Name', () => expect(DlUiProgressCircle.name).toBe('dl-ui-progress-circle'));

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('slot content', () => {
    const wrapperTest = mount(DlUiProgressCircle, { slots : { default : '<div class="fake-msg"></div>' } });
    const div = wrapperTest.find('div');
    expect(div.findAll('.fake-msg').length).toBe(1);
  });

  describe('percentValue', () => {
    it('value', async () => {
      const wrapperTest = mount(DlUiProgressCircle,
        {
          propsData : {
            min   : 0,
            max   : 50,
            value : 0
          }
        });
      await wrapperTest.setProps({ value : 25 });
      await nextTick();

      expect(wrapperTest.vm.percentValue).toBe(50);
      await wrapperTest.setProps({ value : 50 });

      await nextTick();
      expect(wrapperTest.vm.percentValue).toBe(100);
    });
  });

  it('over max', async () => {
    const wrapperTest = mount(DlUiProgressCircle,
      {
        propsData : {
          min   : 0,
          max   : 10,
          value : 0
        }
      });
    await wrapperTest.setProps({ value : 11 });
    await nextTick();
    expect(wrapperTest.vm.percentValue).toBe(100);
  });

  it('sub min max', async () => {
    const wrapperTest = mount(DlUiProgressCircle,
      {
        propsData : {
          min   : 5,
          max   : 10,
          value : 0
        }
      });
    await wrapperTest.setProps({ value : 4 });
    await nextTick();
    expect(wrapperTest.vm.percentValue).toBe(0);

  });

});
