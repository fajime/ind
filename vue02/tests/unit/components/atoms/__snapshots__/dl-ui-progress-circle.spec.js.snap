// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-progress-circle renders correctly 1`] = `
<div
  class="dl-ui-progress-circle"
>
  <div
    class="dl-ui-progress-circle__circle"
  >
    <svg
      class="dl-ui-progress-circle__svg"
      viewbox="0 0 10 10"
    >
      <circle
        class="dl-ui-progress-circle__progress-base"
        cx="50%"
        cy="50%"
        r="46%"
      />
      <circle
        class="dl-ui-progress-circle__progress-fill"
        cx="50%"
        cy="50%"
        r="46%"
        style="--percent: 0;"
      />
    </svg>
    <div
      class="dl-ui-progress-circle__inner-content"
    >
      
      
    </div>
  </div>
</div>
`;
